### PR TITLE
[DROP-IN] Fixed alternative route selection.

### PR DESCRIPTION
**Pull request by tomaszrybakiewicz**
Closes [#1618](https://github.com/mapbox/navigation-sdks/issues/1618)

### Description

- Fixed alternative route selection by avoiding unnecessary route fetch in `InfoPanelHeaderComponent`

### Screenshots or Gifs

_QA App screen cap on Pixel 2_

https://user-images.githubusercontent.com/2678039/162825783-df87144f-fbed-4816-9c0b-38974f996b73.mp4

### CODE REVIEW

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/infopanel/InfoPanelHeaderComponent.kt [hunk 1]
```diff
@@ -71,32 +80,28 @@ internal class InfoPanelHeaderComponent(
...
-                val isRouteReady = waitForFetched()
-                if (isActive && isRouteReady) {
-                    navigationStateViewModel.invoke(
-                        NavigationStateAction.Update(navigationState)
-                    )
-                } else {
-                    logW(TAG, "Routes are not ready")
-                }
-            }
-        } ?: logE(TAG, "Cannot fetch routes because state is incorrect")
```
Comment (kmadsen) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/infopanel/InfoPanelHeaderComponent.kt:91]: Does it silently fail now? I'm not seeing any error/warning logs from RoutesViewModel or this component

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/infopanel/InfoPanelHeaderComponent.kt [hunk 2]
```diff
@@ -71,32 +80,28 @@ internal class InfoPanelHeaderComponent(
...
+            waitForFetched()
+        } ?: false
     }
 
     private suspend fun waitForFetched(): Boolean {
```
Comment (kmadsen) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/infopanel/InfoPanelHeaderComponent.kt:103]: nitpick: Just noticed this, seems better to say waitForReady. Up to you

```suggestion
    private suspend fun waitForReady(): Boolean {
```

**Review by kmadsen — APPROVED** [review]
Small comments, but LGTM 👍

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/infopanel/InfoPanelHeaderComponent.kt (hunk 2 above)
Comment (tomaszrybakiewicz) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/infopanel/InfoPanelHeaderComponent.kt]: Done

**Pull request merged**